src/interface/meshblock_data.cpp

```diff
@@ -320,12 +330,19 @@ MeshBlockData<T>::GetVariablesByFlag(const Metadata::FlagCollection &flags,
 
 template <typename T>
 typename MeshBlockData<T>::VarList
-MeshBlockData<T>::GetVariablesByUid(const std::vector<Uid_t> &uids, bool flux) {
+MeshBlockData<T>::GetVariablesByUid(const std::vector<Uid_t> &uids,
+                                    const FluxRequest flux) {
+  PARTHENON_INSTRUMENT
   typename MeshBlockData<T>::VarList var_list;
   for (auto i : uids) {
     auto v = GetVarPtr(i);
-    if (flux) {
-      const auto &vf = varMap_.at(v->metadata().GetFluxName());
+    if (flux == FluxRequest::OnlyFlux) {
+      const auto &m = v->metadata();
+      if (!m.IsSet(Metadata::WithFluxes)) {
+        PARTHENON_FAIL("Flux of var " + v->label() +
+                       " requested, but var does not have fluxes.");
+      }
+      const auto &vf = varMap_.at(m.GetFluxName());
       var_list.Add(vf);
     } else {
       var_list.Add(v);
```

Comment:
Should there be a check here to ignore variables if `v->IsSet(Metadata::Flux)` if `FluxRequest::NoFlux`?

Reply:
Yes. Added now. There's a continue statement above this line.